lib/rubocop/cop/bundler/gem_version.rb

```diff
@@ -45,6 +45,21 @@ class GemVersion < Base
           (send nil? :gem <(str #version_specification?) ...>)
         PATTERN
 
+        # @!method with_git_ref?(node)
+        def_node_matcher :with_git_ref?, <<~PATTERN
+          (send nil? :gem <(hash <#git? #tag_ref? ...>) ...>)
+        PATTERN
+
+        # @!method tag_ref?(node)
+        def_node_matcher :tag_ref?, <<~PATTERN
+          (pair (sym {:tag :ref}) (str _))
+        PATTERN
+
+        # @!method git?(node)
+        def_node_matcher :git?, <<~PATTERN
```

Comment:
Do we really need to change for the VCS? I doubt someone would use `:tag` or `:ref` without them and this will probably result in some mistake. Haven't tried it, though.